### PR TITLE
ffi: Use async functions in `AuthenticationService`

### PR DESCRIPTION
**Pull request by zecakeh**
I believe that all async methods called from AuthenticationService's API are now ported away from `RUNTIME.block_on`.

### CODE REVIEW

File: bindings/matrix-sdk-ffi/src/authentication_service.rs [hunk 1]
```diff
@@ -40,7 +40,7 @@ pub struct AuthenticationService {
     base_path: String,
     passphrase: Option<String>,
     user_agent: Option<String>,
-    client: RwLock<Option<Client>>,
+    client: tokio::sync::RwLock<Option<Client>>,
```
Comment (bnjbvr) [inline bindings/matrix-sdk-ffi/src/authentication_service.rs:43]: instead, can we use import renamings please (and remove all qualified paths with `tokio::sync` introduced in this patch)? `StdRwLock` for the libstd one, `AsyncRwLock` for the tokio one.

File: bindings/matrix-sdk-ffi/src/client.rs [hunk 2]
```diff
@@ -347,9 +345,8 @@ impl Client {
     }
 
     /// Gets information about the owner of a given access token.
-    pub(crate) fn whoami(&self) -> anyhow::Result<whoami::v3::Response> {
-        RUNTIME
-            .block_on(async move { self.inner.whoami().await.map_err(|e| anyhow!(e.to_string())) })
+    pub(crate) async fn whoami(&self) -> anyhow::Result<whoami::v3::Response> {
+        self.inner.whoami().await.map_err(|e| anyhow!(e.to_string()))
```
Comment (bnjbvr) [inline bindings/matrix-sdk-ffi/src/client.rs:349]: Preexisting, but isn't it the same as:

```suggestion
        Ok(self.inner.whoami().await?)
```

**Review by bnjbvr — APPROVED** [review]
Thanks.

**Comment by jmartinesp** [comment]
It seems to work fine on EX Android.

File: bindings/matrix-sdk-ffi/src/client.rs (hunk 2 above)
Comment (zecakeh) [reply bindings/matrix-sdk-ffi/src/client.rs]: Applied

**Pull request merged**